### PR TITLE
Don't bring down the entire logging if something happens when adding to log

### LINKED ISSUE #3: Exception when appending log brings down all logging
```
Application: appfactoryservice.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.IO.IOException
Stack:
   at System.IO.__Error.WinIOError(Int32, System.String)
   at System.IO.FileStream.Init(System.String, System.IO.FileMode, System.IO.FileAccess, Int32, Boolean, System.IO.FileShare, Int32, System.IO.FileOptions, SECURITY_ATTRIBUTES, System.String, Boolean, Boolean, Boolean)
   at System.IO.FileStream..ctor(System.String, System.IO.FileMode, System.IO.FileAccess, System.IO.FileShare, Int32, System.IO.FileOptions, System.String, Boolean, Boolean, Boolean)
   at System.IO.StreamWriter..ctor(System.String, Boolean, System.Text.Encoding, Int32, Boolean)
   at System.IO.StreamWriter..ctor(System.String, Boolean, System.Text.Encoding, Int32)
   at System.IO.StreamWriter..ctor(System.String, Boolean)
   at System.IO.File.AppendText(System.String)
   at eCraft.appFactory.appFactoryService.Logger.AppendToLog(System.String, System.String)
   at eCraft.appFactory.appFactoryService.Logger.LogHelper(System.String, Int32, System.String, System.String, System.String)
   at eCraft.appFactory.appFactoryService.Logger.LogStandardOutput(System.String, Int32, System.String, System.String)
   at System.Diagnostics.AsyncStreamReader.FlushMessageQueue()
   at System.Diagnostics.AsyncStreamReader.GetLinesFromStringBuilder()
   at System.Diagnostics.AsyncStreamReader.ReadBuffer(System.IAsyncResult)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.IO.Stream+ReadWriteTask.System.Threading.Tasks.ITaskCompletionAction.Invoke(System.Threading.Tasks.Task)
   at System.Threading.Tasks.Task.FinishContinuations()
   at System.Threading.Tasks.Task.Finish(Boolean)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(System.Threading.Tasks.Task ByRef)
   at System.Threading.Tasks.Task.ExecuteEntry(Boolean)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
```

### PR DESCRIPTION
**Pull request by sorenhoglund**
Fixes: https://github.com/ecraft/ecraft-watchdogservice/issues/3

### CODE REVIEW

File: eCraft.appFactory.appFactoryService/Logger.cs [hunk 1]
```diff
@@ -61,15 +63,31 @@ private static void AppendToLog(string logFileName, string message)
...
+                    WriteEventLog(e);
                 }
             }
         }
 
+        private static void WriteEventLog(Exception e)
+        {
+            if (!EventLog.SourceExists(EventSourceName))
+            {
+                EventLog.CreateEventSource(EventSourceName, "Application");
```
Comment (sdahlbac) [inline eCraft.appFactory.appFactoryService/Logger.cs:86]: does it have permission to create an event source?

**Review by sdahlbac — APPROVED** [review]
one potential issue, but apart from that it makes sense.

File: eCraft.appFactory.appFactoryService/Logger.cs (hunk 1 above)
Comment (sorenhoglund) [reply eCraft.appFactory.appFactoryService/Logger.cs:86]: @sdahlbac the manifest says 
`<requestedExecutionLevel  level="requireAdministrator" uiAccess="false" />`

so it should be alright

Comment (sdahlbac) [reply eCraft.appFactory.appFactoryService/Logger.cs:86]: ok, let's hope so

**Comment by perlun** [comment]
@sorenhoglund Thanks for the fix. :+1:

**Pull request merged**